debezium-core/src/main/java/io/debezium/connector/base/ChangeEventQueue.java

```diff
@@ -154,6 +171,7 @@ public void enqueue(T record) throws InterruptedException {
                 metronome.pause();
                 LOGGER.debug("checking for more records...");
             }
+            currentQueueSizeInBytes.addAndGet(-ObjectSizeCalculator.getObjectSize(records));
```

Comment:
@vanhoale Would it be possible to handle queue size set to zero as a special value that disables the rate limitng and calculation of the  object size?

Reply:
@vanhoale
`Field::isPositiveLong` would become `Field::isNonNegativeLong` and description will be updated, default value will be `0`
this line will be encapusalted in `If (maxQueueSizeInBytes > 0)`
So if the config value is not set or is set to zero the calculation will not be triggered. The waiting loop above will wait only for non zero maxQueueSizeInBytes

Reply:
is it returned DEFAULT_MAX_QUEUE_SIZE_IN_BYTES if I don't pass the max.queue.size.in.bytes variable?

Reply:
Yes and we want to disable queue size limitng by default, hence choosign the 0.

Reply:
ah, got your point, so you want a logic like: if we don't pass the maxQueueSizeInBytes then we shouldn't use ObjectSizeCalculator.getObjectSize() method?

Reply:
Exactly!

Reply:
I like it, will update it